### PR TITLE
Version 5.1.3

### PR DESCRIPTION
**Pull request by mbrentini**
- Multishop bug fix in hookUpdateCarrier
- Bug fix in PS 1.5+ OPC template file
- Improved order address rewriting function
- Improved verification of Pickup point selection
- Slight CSS fixes
- Beautify code

### CODE REVIEW

**Comment by mbrentini** [comment]
Hi @jeromenadaud / @Quetzacoalt91 / @tchauviere ,

You will find an update for the dpdfrance module, aiming to fix some bugs. You will be pleased to know that I try to reduce the amount of commits :)

Thank you for your help !
Marco

**Comment by gaillafr** [comment]
Hi @mbrentini, once the DPD France module is installed I don't have any carriers anymore:
https://www.dropbox.com/s/mn8dfhud9mqh6pu/Screenshot%202015-06-08%2013.32.57.png?dl=0

Even if the module is just install, not configured.

**Comment by mbrentini** [comment]
Hi @gaillafr ,

When installing the module, it creates a "France" delivery zone. So make sure your carriers are affected to this new zone and that a shipping rate is set.

**Comment by gaillafr** [comment]
Thank you for your quick reply, but the problem is, I did not do anything. I just installed your module.
I don't want our merchant to have to reconfigure all of their carriers.

The module should not alter the way the merchant configured his previous carriers.

**Comment by mbrentini** [comment]
This situation almost never happens, as the very large majority of merchants already have a “France” zone, created by themselves or by the So Colissimo module.

During installation, if our module detects an existing “France” zone, we will reuse that zone, so the merchants don’t have to configure again their previous carriers.

See : https://github.com/DPDFrance/DPDFrance_PrestaShop1.0-1.6/blob/master/dpdfrance.php#L329-L357

Bien cordialement,
Marco Brentini
Ingénieur Technico-Commercial | Technical Sales Engineer
+33 (0)6 30 61 19 02 | marco.brentini@dpd.fr<mailto:marco.brentini@dpd.fr>
––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––
[cid:image001.gif@01D0A1F5.B532A900]
DPD Etablissement 975
27 Rue du Colonel Pierre Avia, 75015 PARIS | dpd.fr<http://www.dpd.fr/>
[cid:image002.gif@01D0A1F5.B532A900]
––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––––
DPD France SAS – Capital Social : 18.500.000 €. 444 420 830 RCS Paris
Siège Social : 27 rue du Colonel Pierre Avia 75015 Paris
Numéro TVA intracommunautaire : FR24 444 420 830

De : François Gaillard [mailto:notifications@github.com]
Envoyé : lundi 8 juin 2015 14:11
À : PrestaShop/dpdfrance
Cc : BRENTINI Marco
Objet : Re: [dpdfrance] Version 5.1.3 (#21)


Thank you for your quick reply, but the problem is, I did not do anything. I just installed your module.
I don't want our merchant to have to reconfigure all of their carriers.

The module should not alter the way the merchant configured his previous carriers.

**Comment by gaillafr** [comment]
According to the stats we can obtain on the Cloud version of PrestaShop, even France based shop, merchants almost *never* create a "France" zone. It's not automatic. For most of them they don't even know how to do and they don't have SoColissimo neither.

**Comment by mbrentini** [comment]
François,

The fact is that our Predict and Pickup delivery services are limited to the French territory only. We *must* disable these carriers for customers coming from other countries than France. 

As I told you earlier, in the large majority of cases, our French merchants on Prestashop "download version" have a "France" delivery zone.

As we provide the technical support of our module for our customers, we know how to advise our customers facing this situation, even if there are very few cases ! Our documentation also explains this situation in the "Troubleshooting" section.

The only drawback of our method is that if there is no "France" zone, the merchant has to report his shipping rates for the other carriers, but this should not take very much time. Merchants usually don't have more than 4~5 carriers for France.

I have found 2 other ways to restrict the carriers to France but the method and results don't satisfy us...

- Using javascript, in hookExtracarrier, if customer iso code is not 'FR' or 'FX', then load a template file which will remove the carrier. If a javascript file throws an error, or if the javascripts are disabled by the client's browser, this method is not valid. For example, this is used by the SoColissimo module.

- Using the getOrderShippingCost functions, if the customer iso code is not 'FR' or 'FX', then return a 'false' cost, thus disabling the carrier. This method is great but is not valid for Prestashop 1.3.x and lower. This is used by the Chronopost module.

Our module is perfectly compatible with all versions of PS, from 1.0 to 1.6, which is a great opportunity for our company, and for our customers who don't want to migrate (and we have still maaaaaaany customers on PS 1.3 and 1.2.5)
It is also a plus for me, as I don't have to maintain multiple iterations of this module.

If you know a method to restrict the Pickup and Predict carriers to the French territory, compatible with all PS versions, and not using Javascript, please don't hesitate to advise us :)

Thank you
Marco

**Comment by gaillafr** [comment]
Hi Marco, OK but we do not agree with the fact the module is changing any behaviour.
Also after trying to install the module on PrestaShop 1.3 nothing appears anymore in  the 'Modules' page, so if the module only works on PS 1.4 to 1.6 checking if the address is FR would be enough:
https://www.dropbox.com/s/6yb9xl93wwbgmu1/Screenshot%202015-06-08%2016.09.38.png?dl=0

Thank you,
Kind regards.

**Comment by mbrentini** [comment]
The module has been tested successfully on 1.2.5, 1.3.7, 1.4.11.0, 1.5.6.2, 1.6.x

I don't have any error or white page on my local PS 1.3.7. 

image

Would you please enable debug mode to see the error message ?

File: dpdfrance.php [hunk 1]
```diff
@@ -753,20 +718,42 @@ public function hooknewOrder($params)
...
+			Configuration::updateValue('DPDFRANCE_PREDICT_CARRIER_LOG', Configuration::get('DPDFRANCE_PREDICT_CARRIER_LOG', null, $shop['id_shop_group'], $shop['id_shop']).'|'.(int)$params['carrier']->id, false, $shop['id_shop_group'], $shop['id_shop']);
 		}
-		if ((int)$params['id_carrier'] == (int)Configuration::get('DPDFRANCE_CLASSIC_CARRIER_ID'))
+		if ((int)$params['id_carrier'] == (int)Configuration::get('DPDFRANCE_CLASSIC_CARRIER_ID', null, $shop['id_shop_group'], $shop['id_shop']))
 		{
-			Configuration::updateValue('DPDFRANCE_CLASSIC_CARRIER_ID', (int)$params['carrier']->id);
-			Configuration::updateValue('DPDFRANCE_CLASSIC_CARRIER_LOG', Configuration::get('DPDFRANCE_CLASSIC_CARRIER_LOG').'|'.(int)$params['carrier']->id);
+			Configuration::updateValue('DPDFRANCE_CLASSIC_CARRIER_ID', (int)$params['carrier']->id, false, $shop['id_shop_group'], $shop['id_shop']);
+			Configuration::updateValue('DPDFRANCE_CLASSIC_CARRIER_LOG', Configuration::get('DPDFRANCE_CLASSIC_CARRIER_LOG', null, $shop['id_shop_group'], $shop['id_shop']).'|'.(int)$params['carrier']->id, false, $shop['id_shop_group'], $shop['id_shop']);
```
Comment (Quetzacoalt91) [inline dpdfrance.php]: I guess you're trying to handle the multishop case here. You should add a test at the beginning with `Shop::isFeatureActive()`. If `true`, you execute the first part, and if `false` you execute the second one. But you do not need to execute both.

**Comment by Quetzacoalt91** [comment]
Hi,

Can you please a updated file config.xml ?

Thanks

**Comment by mbrentini** [comment]
Hi @Quetzacoalt91 

I just added config.xml an added a test (Shop::isFeatureActive()) in hookupdateCarrier as asked.

Thank you for your review.

Marco

**Comment by Quetzacoalt91** [comment]
Hi @mbrentini,

Thanks, your PR has just been merged and the changes will be available from tomorrow.

Best regards

**Comment by mbrentini** [comment]
Hi @Quetzacoalt91 

Thank you very much :+1: 

Kind regards
Marco

**Pull request merged**